tools/azsdk-cli/Azure.Sdk.Tools.Cli/Services/Languages/DotNetLanguageSpecificChecks.cs

```diff
@@ -12,19 +10,246 @@ namespace Azure.Sdk.Tools.Cli.Services;
+    private async ValueTask<CLICheckResponse> VerifyDotnetVersion()
+    {
+        var dotnetSDKCheck = await _processHelper.Run(new ProcessOptions(DotNetCommand, ["--list-sdks"]), CancellationToken.None);
+        if (dotnetSDKCheck.ExitCode != 0)
+        {
+            _logger.LogError(".NET SDK is not installed or not available in PATH");
+            return new CLICheckResponse(dotnetSDKCheck.ExitCode, $"dotnet --list-sdks failed with an error: {dotnetSDKCheck.Output}");
+        }
+
+        var dotnetVersions = dotnetSDKCheck.Output.Split(Environment.NewLine.ToCharArray(), StringSplitOptions.RemoveEmptyEntries);
+        var latestVersionNumber = dotnetVersions[dotnetVersions.Length - 1].Split('[')[0].Trim();
+
+        if (Version.TryParse(latestVersionNumber, out var installedVersion) &&
+            Version.TryParse(RequiredDotNetVersion, out var minimumVersion))
+        {
+            if (installedVersion >= minimumVersion)
+            {
+                _logger.LogInformation(".NET SDK version {InstalledVersion} meets minimum requirement of {RequiredVersion}", latestVersionNumber, RequiredDotNetVersion);
+                return new CLICheckResponse(0, $".NET SDK version {latestVersionNumber} meets minimum requirement of {RequiredDotNetVersion}");
+            }
+            else
+            {
+                _logger.LogError(".NET SDK version {InstalledVersion} is below minimum requirement of {RequiredVersion}", latestVersionNumber, RequiredDotNetVersion);
+                return new CLICheckResponse(1, "", $".NET SDK version {latestVersionNumber} is below minimum requirement of {RequiredDotNetVersion}");
+            }
+        }
+        else
+        {
+            _logger.LogError("Failed to parse .NET SDK version: {VersionString}", latestVersionNumber);
+            return new CLICheckResponse(1, "", $"Failed to parse .NET SDK version: {latestVersionNumber}");
+        }
+    }
+
+    private string? GetServiceDirectoryFromPath(string packagePath)
+    {
+        string? serviceDirectory = null;
+        var normalizedPath = packagePath.Replace('\\', '/');
+        var sdkIndex = normalizedPath.IndexOf("/sdk/", StringComparison.OrdinalIgnoreCase);
+
+        if (sdkIndex >= 0)
+        {
+            var pathAfterSdk = normalizedPath.Substring(sdkIndex + 5); // Skip "/sdk/"
+            var segments = pathAfterSdk.Split('/', StringSplitOptions.RemoveEmptyEntries);
+            if (segments.Length > 0)
+            {
+                serviceDirectory = segments[0];
+                _logger.LogDebug("Extracted service directory: {ServiceDirectory}", serviceDirectory);
+            }
+            else
+            {
+                _logger.LogDebug("No segments found after /sdk/ in path");
+            }
+        }
+        else
+        {
+            _logger.LogDebug("Path does not contain /sdk/ segment");
+        }
+
+        return serviceDirectory;
+    }
+
+    private string? GetPackageNameFromPath(string packagePath)
```

Comment:
Only fyi here is that this is used in ValidateChangelog -- https://github.com/Azure/azure-sdk-tools/blob/906f82a6b1d49cb333d6edfb18883ddeb3557897/tools/azsdk-cli/Azure.Sdk.Tools.Cli/Services/Languages/LanguageChecks.cs#L269